src/app/component/ProductCard/ProductCard.component.js

```diff
@@ -101,12 +102,31 @@ export class ProductCard extends PureComponent {
     }
 
     renderProductPrice() {
-        const { product: { price_range } } = this.props;
+        const {
+            product: { price_range, type_id },
+            isConfigurableProductOutOfStock,
+            isBundleProductOutOfStock
+        } = this.props;
 
         if (!price_range) {
             return <TextPlaceholder />;
         }
 
+        switch (type_id) {
+        case CONFIGURABLE:
+            if (isConfigurableProductOutOfStock()) {
+                return null;
+            }
+            break;
+        case BUNDLE:
+            if (isBundleProductOutOfStock()) {
+                return null;
+            }
+            break;
+        default:
+            break;
+        }
```

Comment:
Actually my proposal of switch was bad idea, in the perfect scenario we change this to a typeMap. But I am OK with that for now.